### DISCUSSION TITLE
A very quick pass for testing purposes

### DISCUSSION THREAD

**Discussion by brutlern** [PartialVolume/shredos.x86_64#145] [Q&A] [answered]
I am new to using this software and would like to know if there is an option to do a very quick run for testing purposes. I am currently using verify_zero method, but that one is still 8 minutes on a 256GB drive. So if I am testing --autopoweroff for example, I have to wait 8 minutes everytime, to check the behavior on each machine I run the test.

**Comment by PartialVolume** [comment]
I create one or more virtual discs.

- In ShredOS switch to the virtual terminal ALT-F2.
- Create a small virtual disc, this resides in CPU memory so you need to create a suitable size. If your CPU memory is 16GBytes you don't want to create a 32GByte drive So as an example, create a 1Gbyte file using the command `truncate -s 1G loopbackfile001.img`
- Create the virtual disc using the command `losetup -fP loopbackfile001.img`
- display the loopback device name `losetup -a | grep -i loopbackfile`
```
losetup -a | grep -i loopbackfile
/dev/loop28: [66306]:14027243 (/home/nick/Data/virtual_disk/loopbackfile001.img)
```

- Using the device name found in the previous step, run nwipe in the same virtual terminal (don't use the main nwipe in ALT-F1), stay in ALT-F2 `nwipe /dev/loop28`

> **Warning**
You need to specify a virtual device on the command line, you can't just type `nwipe` on it's own as it won't display a virtual device.

![Screenshot_20230511_110541](https://github.com/PartialVolume/shredos.x86_64/assets/22084881/d7d8ed35-ce75-4016-b861-fa8c52563a09)

**Reply by PartialVolume** [reply]
If testing --autopoweroff  you would need to edit grub and put the loopdevice in nwipe options. The issue though is you then create the loop device after you have specified the loop device name on the command line so unless you are lucky it won't display the loop device in the main virtual terminal unless they match.

So I'm not sure how much use the above method would be in your specific use case. I can add an option that would show loop devices in nwipe i.e `nwipe --loop` then it would simply be a case of editing the linux command line in grub and adding --loop so that nwipe shows loop devices.

**Reply by brutlern** [reply]
Yes, under ALT-F2, works fine, a small 1G drive which can be "erased" in a matter of seconds, but I would need it to be visible in ALT-F1. I have dozens of different type of machines (from 15 year old Pentiums to modern machines, old PATA drives to Gen5 NVMEs), and I need to quickly test on which of them ShredOS works as intented (specifically, unattended operation) and which have issues. I quickly need to test if they boot, autonuke, save logs, and shutdown without user intervention (except plug in the USB drive and start the machine of course).

**Reply by PartialVolume** [reply, marked as answer]
So really what you need is a option called, say for instance `--test` that you can add to nwipe options in the grub file.

ShredOS/nwipe would parse the options and if finding test, create a couple of virtual drives and then call nwipe in ALT-F1 specifying those drives. It wouldn't take much to modify the launcher in ShredOS to do that. I would probably add the ability to specify the virtual disc size, as 1GB would be too big for old systems. So the option syntax could be something like `--test=500M/2` etc.. Using that command it would create two 500MB virtual discs.

Does that sound like something that would be useful.

Finally, when specifying --test would you want ShredOS/nwipe to display all the system drives as well as the virtual drives? I'm guessing probably yes to that?

**Reply by brutlern** [reply]
Yes, that's exactly it. Of course, don't do this just for my sake. Maybe if this is an option that others might also need/want/be interested in.

**Reply by PartialVolume** [reply]
In the end I added a simpler solution, if you add /dev/loop0 and/or /dev/loop1 to nwipe_options in grub.cfg then ShredOS will automatically create one or two virtual drives that are 1MByte in size. At the moment you can't edit the size of these virtual drives but hopefully they will be sufficient for most testing. I can always come back to this if something a bit better is required. commit [7734422](https://github.com/PartialVolume/shredos.x86_64/commit/77344223ce8725ef2d3ff452fc454688e3b81c06). Pre-released in [v2021.08.2_23.3_x86-64_0.34](https://github.com/PartialVolume/shredos.x86_64/releases/tag/v2021.08.2_23.3_x86-64_0.34)

**Comment by PartialVolume** [comment]
Closing discussion for now, please fell free to reopen if the current solution turns out to be insufficient for your testing.
